teos/src/cli.rs

```diff
@@ -75,20 +75,28 @@ async fn main() {
             println!("{}", pretty_json(&appointments.into_inner()).unwrap());
         }
         Command::GetAppointments(appointments_data) => {
-            match Locator::from_hex(&appointments_data.locator) {
-                Ok(locator) => {
-                    match client
-                        .get_appointments(Request::new(msgs::GetAppointmentsRequest {
-                            locator: locator.to_vec(),
-                        }))
-                        .await
-                    {
-                        Ok(appointments) => {
-                            println!("{}", pretty_json(&appointments.into_inner()).unwrap())
+            match appointments_data
+                .user_id
+                .map(|id| UserId::from_str(&id).map(|user_id| user_id.to_vec()))
+                .transpose()
+            {
+                Ok(user_id) => match Locator::from_hex(&appointments_data.locator) {
+                    Ok(locator) => {
+                        match client
+                            .get_appointments(Request::new(msgs::GetAppointmentsRequest {
+                                locator: locator.to_vec(),
+                                user_id,
+                            }))
+                            .await
+                        {
+                            Ok(appointments) => {
+                                println!("{}", pretty_json(&appointments.into_inner()).unwrap())
```

Comment:
I'm wondering if, if both user_id and locator are provided, we should return a single item instead of a list with a single item (notice a locator-uuid pair can only map a single item).

In case we want this, it can be left for a followup

Reply:
I think Uniformity of return type is a nice to have, specifically for simplification, what do you think ?

Reply:
I generally agree